### DISCUSSION TITLE
How to perform secondary action for upsert in Javascript

### DISCUSSION THREAD

**Discussion by notKvS** [supabase/supabase#12006] [Questions] [answered]
I have table in which I want to insert data if certain id for row is not there, and if it is there I want to update the row with new data.
Something like this
```
INSERT INTO the_table (id, column_1, column_2) 
VALUES (1, 'A', 'X'), (2, 'B', 'Y'), (3, 'C', 'Z')
ON CONFLICT (id) DO UPDATE 
  SET column_1 = excluded.column_1, 
      column_2 = excluded.column_2;
```

I am using upsert, but unable to figure out how to perform update if `onConflict` occurs for certain row

**Comment by GaryAustin1** [comment, marked as answer]
You add the {onConflict: 'col'} to the upsert in js.
Then the column must be declared unique (in the UI you can check a box, or create a unique constraint in SQL).

**Reply by notKvS** [reply]
Thanks for the answer, what I was doing wrong was I didn't declared that column as unique (the one I was using in onConflict).
